apps/api/src/applicant/ienapplicant.controller.ts

```diff
@@ -303,4 +303,34 @@ export class IENApplicantController {
       }
     }
   }
+
+  @ApiOperation({
+    summary: 'Get Specific Applicant job and related milestone',
+  })
+  @UseInterceptors(ClassSerializerInterceptor)
+  @ApiResponse({ status: HttpStatus.OK, type: EmptyResponse })
+  @HttpCode(HttpStatus.OK)
+  @RouteAcceptsRoles(
+    ValidRoles.HEALTH_AUTHORITY,
+    ValidRoles.HEALTH_MATCH,
+    ValidRoles.MINISTRY_OF_HEALTH,
+  )
+  @Get('/:id/job/:job_id')
+  getApplicantJob(
+    @Param('id') id: string,
+    @Param('job_id') job_id: string,
+  ): Promise<ApplicantJobRO | undefined> {
+    try {
+      return this.ienapplicantService.getApplicantJob(id, job_id);
+    } catch (e) {
+      if (e instanceof NotFoundException) {
+        throw e;
+      } else if (e instanceof QueryFailedError) {
+        throw new BadRequestException(e);
+      } else {
+        // statements to handle any unspecified exceptions
+        throw new InternalServerErrorException('An unknown error occured while fetching ');
```

Comment:
In this way, it silences the original exception. How about logging it or including its message in a new exception?